docs/setup_local_development.md

```diff
@@ -0,0 +1,151 @@
+1. Navigate to the directory `phdi-google-cloud/terraform/implementation`.
+1. Run `terraform init` to initialize the implementation Terraform. It will prompt you for the name of a bucket to store Terraform's state. Enter the name of the bucket output in the earlier step. It will also prompt you for a Project ID and region/zone.
+1. Create a new workspace called "dev" by running `terraform workspace new dev`.
+1. Now you can run `terraform plan` to have Terraform determine the difference between the code locally and the infrastructure currently deployed in GCP. Terraform will return a list of changes, resources it will create, destroy, or modify, that it would make if you chose to move forward with a deployment. After a carefull review, if these changes are acceptable you may deploy them by running `terraform apply`. Please refer to the [Terraform CLI documentation](https://www.terraform.io/cli/commands) for further information on using Terraform locally.
+
+### Continuous Integration and Continuous Deployment (CI/CD)
+
+We have implemented CI/CD pipelines with [GitHub Actions](https://docs.github.com/en/actions) orchestrated by [GitHub Workflows](https://docs.github.com/en/actions/using-workflows/about-workflows) found in the `phdi-google-cloud/.github/` directory.
+
+#### Continuous Integration (CI)
+
+The entire CI pipeline can be found in `phdi-google-cloud/.github/test.yaml`. It runs every time a Pull Request is opened and whenever additional changes are pushed to a branch. Currently, the following steps are included in the CI pipeline:
+
+1. Identify all directories containing a Cloud Function.
+2. Run the unit tests for each Cloud Function.
+3. Check that all Python code complies with Black and Flake8.
+4. Check that all Terraform code is formated properly.
+
+#### Continuous Deployment (CD)
+
+A separate CD pipeline is configured for each GCP environemnt we deploy to. Each of these pipelines is defined in a YAML file starting with "deploy" in the `workflows/` directory (e.g. `phdi-google-cloud/.github/deployment.yaml`). Generally, these pipelines run every time code is merged into the `main` branch of the repository. However, additional dependencies can be specified. For example, a successful deployment to a development environment could required before deploying to a production environment proceeds. When these pipelines run they first look for differences in the infrastructure that is specified in the respository and currently deployed to a given GCP project. If differences are detected, they are resolved by making changes to GCP project to bring into alignment with the repository. In order to grant the GitHub repository permission to make these changes, follow [these instructions](https://github.com/google-github-actions/auth#setup) to authenicate it with GCP.
```

Comment:
@gordonfarrell I understand why you moved CI/CD into this doc. However, there are parts of at least the CD process that are important for the Implementation Guide and by association getting_started. I guess what I'm getting at is that even if someone is just a user of the Starter Kit, but not doing development on it, they may need the CD pipeline (and possibly CI??). Now I'm not quite sure where this content should live...